lnst/Recipes/ENRT/MeasurementGenerators/ForwardingMeasurementGenerator.py

```diff
@@ -0,0 +1,75 @@
+"""
+Module with generator class for Forwarding Recipe.
+
+Copyright 2025 Red Hat, Inc.
+Licensed under the GNU General Public License, version 2 as
+published by the Free Software Foundation; see COPYING for details.
+"""
+
+__author__ = """
+sdobron@redhat.com (Samuel Dobron)
+"""
+
+from lnst.Recipes.ENRT.MeasurementGenerators.BaseFlowMeasurementGenerator import (
+    BaseFlowMeasurementGenerator,
+)
+from lnst.RecipeCommon.Perf.Measurements.ForwardingMeasurement import (
+    ForwardingMeasurement,
+)
+from lnst.RecipeCommon.Perf.Measurements.BaseFlowMeasurement import BaseFlowMeasurement
+
+
+from lnst.RecipeCommon.Perf.Measurements import Flow as PerfFlow
+from lnst.RecipeCommon.endpoints import EndpointPair, IPEndpoint
+
+
+class ForwardingMeasurementGenerator(BaseFlowMeasurementGenerator):
+    @property
+    def net_perf_tool_class(self):
+        def ForwardingMeasurement_partial(*args, **kwargs):
+            return ForwardingMeasurement(
+                *args, ratep=self.params.ratep, burst=self.params.burst, **kwargs
+            )
+
+        return ForwardingMeasurement_partial
+
+    def generator_cpupin(self, flow_id: int) -> list[int]:
+        """
+        Needs to be round-robin, pktgen doesn't support a generator
+        to be pinned to multiple CPUs. If single cpu is not sufficient,
+        just create more flows with same/different src/dst IPs/ports.
+        """
+        return self._cpupin_based_on_policy(
+            flow_id, self.params.perf_tool_cpu, "round-robin"
+        )
+
+    def _create_perf_flows(
+        self,
+        endpoint_pairs: list[EndpointPair[IPEndpoint]],
+        perf_test: str,
+        msg_size,
+    ) -> list[PerfFlow]:
+        """
+        Same as BaseFlowMeasurementGenerator._create_perf_flows, but without
+        iterating over parallel processes. This is already done by generating
+        multiple perf endpoint IPs with different destination IP in ForwardingRecipe.generate_perf_endpoints.
```

Comment:
> ForwardingRecipe.generate_perf_endpoints

this shouldn't be mentioned here... this is a "mixin" class which can be used independently of a specific recipe.

you could rewrite this to explain that this mixin class has some requirements that need to be implemented by a recipe class.